### PR TITLE
Backup: Implement backup staging flag update API

### PR DESCRIPTION
**Pull request by Initsogar**
Similar to #75856, lets add the necessary logic to implement the new API for marking sites as staging (D107983-code).

## Proposed Changes
* Add necessary handlers and redux logic to implement new `POST rewind/staging/update` API to fetch and mark sites as staging.

## To-do
- [x] Action types (API request life cycle)
- [x] API fetch handler
- [x] Actions
- [x] Reducers
- [x] Selectors 
- ~Data query component (if needed)~

## Next steps
* Implement handlers and redux logic for `GET rewind/staging` endpoint (D107983-code).
* Integrate with the new mark a site as Staging Site Jetpack setting.

## Testing Instructions
This is an isolated change that will be integrated on the Jetpack settings page. So at this point you can ensure the unit tests are passing by using `yarn run test-client client/state/rewind`.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/data-layer/wpcom/sites/rewind/staging/update/index.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		},
+		action
+	);
+
+const updateStagingFlagSuccess = ( { siteId }: UpdateStagingFlagRequestActionType ) => [
+	{
+		type: JETPACK_BACKUP_STAGING_UPDATE_REQUEST_SUCCESS,
+		siteId,
+	},
+	successNotice( 'Site has been marked as a Staging Site successfully.', {
```
Comment (jwebbdev) [inline client/state/data-layer/wpcom/sites/rewind/staging/update/index.ts]: This messaging reads like it's always staging:true.  If `action.staging` can be true or false (toggle) maybe it should read something like `Site staging status has been successfully updated.`

File: client/state/data-layer/wpcom/sites/rewind/staging/update/index.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const updateStagingFlagError = (
+	{ siteId }: UpdateStagingFlagRequestActionType,
+	error: string
+) => [
+	{
+		type: JETPACK_BACKUP_STAGING_UPDATE_REQUEST_FAILURE,
+		siteId,
+		error,
+	},
+	errorNotice( 'Marking site as a Staging Site failed. Please, try again.', {
```
Comment (jwebbdev) [inline client/state/data-layer/wpcom/sites/rewind/staging/update/index.ts]: This messaging reads like it's always staging:true. If `action.staging` can be true or false (toggle) maybe it should read something like `Updating site staging status failed.  Please, try again.`

**Review by jwebbdev — COMMENTED** [review]
Tests were successful.  Made a comment about some notice wording that we might want to address.

**Comment by Initsogar** [comment]
@jwebbdev Thanks for the feedback. Good catch! I just pushed a new commit fixing the notices messages on success/failure.

**Review by jwebbdev — APPROVED** [review]
LGTM

**Pull request merged**